enterprise_gateway/services/external/k8s_client_factory.py

```diff
@@ -0,0 +1,31 @@
+"""Contains factory to create kubernetes api client instances using a single confguration"""
+import os
+
+from kubernetes import client, config
+from traitlets.config import SingletonConfigurable
+
+
+class KubernetesClientFactory(SingletonConfigurable):
+    """Manages kubernetes client creation from environment variables"""
+    def __init__(self) -> None:
+        """Maintain a single configuration object and populate based on environment"""
+        super().__init__()
+
+    def get_kubernetes_client(self, get_remote_if_available=True) -> client.ApiClient:
+        """Get kubernetes api client with appropriate configuration"""
+        kubernetes_config: client.Configuration = client.Configuration()
+        if os.getenv("KUBERNETES_SERVICE_HOST"):
+            # Running inside cluster
+            if os.getenv('EG_USE_REMOTE_CLUSTER') and get_remote_if_available:
+                kubeconfig_path = os.environ.get('EG_REMOTE_CLUSTER_KUBECONFIG_PATH', '/etc/kube/config')
```

Comment:
Did you pick the `/etc/kube/config` default, or did this come from something existing?

Reply:
I picked that, honestly by random, I realize I forgot to set this as the default in the helm chart as well, but the idea was for the path to be the same in the helm chart and in code